osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -240,6 +257,74 @@ private void load(TextureStore textures)
             }
         }
 
+        // using string literals as there's no proper processing for LocalizeStrings yet
+        public class SupporterRequiredDrawable : CompositeDrawable
+        {
+            private OsuSpriteText supporterRequiredText;
+
+            public SupporterRequiredDrawable()
+            {
+                RelativeSizeAxes = Axes.X;
+                Height = 225;
+                Alpha = 0;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(TextureStore textures)
+            {
+                AddInternal(new FillFlowContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.Y,
+                    AutoSizeAxes = Axes.X,
+                    Direction = FillDirection.Horizontal,
+                    Children = new[]
+                    {
+                        new Sprite
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativeSizeAxes = Axes.Both,
+                            FillMode = FillMode.Fit,
+                            Texture = textures.Get(@"Online/supporter-required"),
+                        },
+                        supporterRequiredText = new OsuSpriteText
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Font = OsuFont.GetFont(size: 16),
+                            Colour = Colour4.White,
+                            Margin = new MarginPadding { Bottom = 10 },
+                        },
+                        createSupporterTagLink(),
+                    }
+                });
+            }
+
+            public void UpdateText(bool playedFilter, bool rankFilter)
```

Comment:
I don't like this design as it doesn't scale (what if 3 more supporter-specific filters are added N months from now? are 3 more `bool`s going to be added?) This should really be a collection of some sort (not 100% sure of what exactly, the immediate candidate is a collection of the localisation keys themselves).